test/unit/org/apache/cassandra/index/sai/cql/ComplexQueryTest.java

```diff
@@ -18,18 +18,44 @@
 
 package org.apache.cassandra.index.sai.cql;
 
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
 import org.apache.cassandra.cql3.UntypedResultSet;
 import org.apache.cassandra.cql3.restrictions.StatementRestrictions;
 import org.apache.cassandra.exceptions.InvalidRequestException;
 import org.apache.cassandra.index.sai.SAITester;
+import org.apache.cassandra.index.sai.SAIUtil;
+import org.apache.cassandra.index.sai.disk.format.Version;
 
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.junit.Assert.assertEquals;
 
+@RunWith(Parameterized.class)
 public class ComplexQueryTest extends SAITester
 {
+
+    @Parameterized.Parameters(name = "version={0}")
+    public static List<Object> data()
+    {
+        return Stream.of(Version.AA, Version.CURRENT, Version.LATEST).map(v -> new Object[]{ v}).collect(Collectors.toList());
```

Comment:
For the `main` branch PR I'd say better to check all (I tested locally and it passes with ALL as expected btw).
```suggestion
        return Stream.of(Version.ALL).flatMap(List::stream).map(v -> new Object[]{ v}).collect(Collectors.toList());
```

Soon we will move to default EC but there will be still some clusters running on something older. Just want to play it safer.  We can also just do it - test on AA and everything beyond DC maybe? Not blocking this merge though